integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r4/TaskFhirResourceProviderIntegrationTest.java

```diff
@@ -0,0 +1,477 @@
+		assertThat(operationOutcome.hasIssue(), is(true));
+	}
+	
+	@Test
+	public void shouldSearchForExistingTasksAsJson() throws Exception {
+		MockHttpServletResponse response = get("/Task").accept(FhirMediaTypes.JSON).go();
+		
+		assertThat(response, isOk());
+		assertThat(response.getContentType(), is(FhirMediaTypes.JSON.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		org.hl7.fhir.r4.model.Bundle results = readBundleResponse(response);
+		
+		assertThat(results, notNullValue());
+		assertThat(results.getType(), equalTo(Bundle.BundleType.SEARCHSET));
+		assertThat(results.hasEntry(), is(true));
+		
+		List<BundleEntryComponent> entries = results.getEntry();
+		
+		assertThat(entries, everyItem(hasProperty("fullUrl", startsWith("http://localhost/ws/fhir2/R3/Task/"))));
+		assertThat(entries, everyItem(hasResource(instanceOf(Task.class))));
+		assertThat(entries, everyItem(hasResource(validResource())));
+	}
+	
+	@Test
+	public void shouldSearchForExistingTaskAsXml() throws Exception {
+		MockHttpServletResponse response = get("/Task").accept(FhirMediaTypes.XML).go();
+		
+		assertThat(response, isOk());
+		assertThat(response.getContentType(), is(FhirMediaTypes.XML.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		org.hl7.fhir.r4.model.Bundle results = readBundleResponse(response);
+		
+		assertThat(results, notNullValue());
+		assertThat(results.getType(), equalTo(Bundle.BundleType.SEARCHSET));
+		assertThat(results.hasEntry(), is(true));
+		
+		List<BundleEntryComponent> entries = results.getEntry();
+		
+		assertThat(entries, everyItem(hasProperty("fullUrl", startsWith("http://localhost/ws/fhir2/R3/Task/"))));
+		assertThat(entries, everyItem(hasResource(instanceOf(Task.class))));
+		assertThat(entries, everyItem(hasResource(validResource())));
+		
+	}
+	
+	@Test
+	public void shouldReturnSortedAndFilteredSearchResultsForTaskAsJson() throws Exception {
+		MockHttpServletResponse response = get("/Task/?status=accepted&_sort=_lastUpdated").accept(FhirMediaTypes.JSON).go();
+		
+		//		assertThat(response, isOk());
+		assertThat(response.getContentType(), is(FhirMediaTypes.JSON.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		Bundle results = readBundleResponse(response);
+		
+		assertThat(results, notNullValue());
+		assertThat(results.getType(), equalTo(Bundle.BundleType.SEARCHSET));
+		assertThat(results.hasEntry(), is(true));
+		
+		List<Bundle.BundleEntryComponent> entries = results.getEntry();
+		
+		assertThat(entries,
+		    everyItem(hasResource(hasProperty("lastUpdated", hasProperty("status", startsWith("accepted"))))));
+		assertThat(entries, containsInRelativeOrder(
+		    //		        hasResource(hasProperty("lastUpdated", hasProperty("givenAsSingleString", containsString("accepted")))),
+		    hasResource(hasProperty("lastUpdated", hasProperty("givenAsSingleString", containsString("rejected"))))));
```

Comment:
So, here it's important to realise what this is (effectively) translating into. So `containsInRelativeOrder` will effectively loop over `entries` (which is a `List` of `Bundle.BundleEntryComponent` objects). Written out it would look something like this:

```java
for (Bundle.BundleEntryComponent entry : entries) {
    ...
}
```

`hasResource()` extracts the resource for the current entry, like calling `entry.getResource()`. `hasProperty()` extracts the property named as the parameter so, e.g., `hasProperty("lastUpdated", ...)` is equivalent to calling `item.getLastUpdated()`. So effectively what this is trying to determine is if:

```java
entries.get(0).getResource().getLastUpdated().getGivenAsSingleString().contains("rejected")
```

returns `true` or not. But the call chain there isn't really going to work. For one thing, the `lastUpdated` property [isn't part of the FHIR Task resource](https://www.hl7.org/fhir/task.html), but rather of the [metadata associated with the Task resource](https://www.hl7.org/fhir/resource.html#Meta), so to get that value, we need to do:

```java
entries.get(0).getResource().getMeta().getLastUpdated()
```

Second, the `lastUpdated` property doesn't have a `givenAsSingleString` property.

Third, what we're really trying to do here is ensure that we get the values back in the order we expected, so we need to check that the values are in order. For the purposes here, it might be better to use [this pattern](https://github.com/openmrs/openmrs-module-fhir2/blob/d6b645ecea315f71b34a301b35f358b331576012/integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r4/ObservationFhirResourceProviderIntegrationTest.java#L437) to verify the dates are in order. So something like:

```java
for (int i = 1; i < entries.size(); i++) {
	assertThat(entries.get(i - 1).getMeta().getLastUpdated(),
	    sameOrAfter(entries.get(i).getMeta().getLastUpdated()));
}
```

Reply:
Thanks @ibacher, Am sorry for late reply on this, so am getting errors when trying to change some codes especially on has resource() and assert that. can i create a pr for this and you look into it